src/trip_state.py

```diff
@@ -0,0 +1,139 @@
+import json
+from datetime import date, datetime
+from dataclasses import dataclass
+from typing import Dict, TypedDict, Optional
+from ddtrace import tracer
+
+from logger import set_up_logging
+from disk import DATA_DIR
+from util import get_current_service_date
+
+logger = set_up_logging(__name__)
+
+
+class TripState(TypedDict):
```

Comment:
My first instinct was to use a dataclass here, but it seems like we're already passing so many dictionaries around. I'm open to thoughts on how we want to store objects like these, and especially shared patterns for (de)serialization to JSON or CSV.